telemetry_package/main.py

```
@@ -1,34 +1,230 @@
+            durability=QoSDurabilityPolicy.VOLATILE,
+            liveliness=QoSLivelinessPolicy.AUTOMATIC
+        )
+
+        # For tracking the timestamps between the last received messages for throttled log-message names
+        self._message_throttle_table = {}
+
+        # Check if the socket already exists
+        if os.path.exists(self.chosen_socket_path):
+            raise FileExistsError('Socket already exists.')
+
+        try:
+            # Create the socket on the given path and listen for one connection
+            self.server_sock = socket.socket(
+                socket.AF_UNIX, socket.SOCK_STREAM)
+            self.server_sock.bind(self.chosen_socket_path)
+            os.chmod(self.chosen_socket_path, 0o777)
+            print(
+                f"Listening for connection on the UNIX-socket at {self.chosen_socket_path} ...")
+            self.server_sock.listen(1)
+
+            # Accept the connection
+            self.client_sock, _ = self.server_sock.accept()
+            print('Connection accepted. Creating subscriptions...')
+
+            # Subscribe to /rosout
+            self.rosout_subscription = self.create_subscription(
+                Log,
+                'rosout',
+                self.__rosout_callback,
+                qos_profile=qos_profile)
+
+            self.rosout_subscription
+            print(f'Created subscription to "{self.namespace}/rosout".')
+
+            # Subscribe to TopicNames.Heartbeat
+            self.heartbeat_subscription = self.create_subscription(
+                Heartbeat,
+                TopicNames.Heartbeat,
+                self.__heartbeat_callback,
+                qos_profile=qos_profile)
+
+            self.heartbeat_subscription
+            print(f'created subscription to "{self.namespace}/heartbeat".')
+
+        except Exception as e:
+            print(f'Error occurred: {e}')
+            self.server_sock.close()
+            os.remove(self.chosen_socket_path)

```

```suggestion
            os.remove(self.chosen_socket_path)
            raise SystemExit

```

---

Ist better to reraise the exception or to call sys.exit()
```suggestion
            os.remove(self.chosen_socket_path)
            raise e
```
```suggestion
            os.remove(self.chosen_socket_path)
            sys.exit(e)
```